src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,331 @@
+digitizer. If you also have idealized positions, where you wish the optodes to be
+placed, these can be listed in the template values
+(for example for "template positions" computed on a sphere).
+SNIRF contains arrays for both
+the 3D and 2D locations of data. The BIDS format MUST store the 3D locations if
+available, and only the 2D locations if 3D positions are unavailable. The storage
+of 2D locations would be indicated by the z field containing an `n/a` value.
+
+The following columns MUST be present:
+
+{{ MACROS___make_columns_table(
+   {
+      "name__optodes": "REQUIRED",
+      "type__optodes": "REQUIRED",
+      "x__optodes": "REQUIRED",
+      "y__optodes": "REQUIRED",
+      "z__optodes": "REQUIRED",
+   }
+) }}
+
+The following columns MAY be present:
+
+{{ MACROS___make_columns_table(
+   {
+      "x__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "y__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "z__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "description__optode": "OPTIONAL",
+      "detector_type": "OPTIONAL",
+      "source__optodes": "OPTIONAL",
+   }
+) }}
+
+Example:
+```Text
+name    type         x          y         z          template_x    template_y   template_z
+A1      source       -0.0707    0.0000    -0.0707    -0.07         0.00         0.07
+Fz      detector     0.0000     0.0714    0.0699     0.0           0.07         0.07
+S1      source       -0.2707    0.0200    -0.1707    -0.03         0.02         -0.2
+D2      detector     0.0022     0.1214    0.0299     0.0           0.12         0.03
+VisS2   source       -0.1707    0.1200    -0.3707    -0.1          0.1          -0.4
+VisD4   detector     0.0322     0.2214    0.2299     0.02          0.22         0.23
+```
+
+## Coordinate System JSON (`*_coordsystem.json`)
+
+{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["coordsystem"]) }}
+
+A `*_coordsystem.json` file is used to specify the fiducials, the location of anatomical landmarks,
+and the coordinate system and units in which the position of optodes and landmarks is expressed.
+Fiducials are objects with a well-defined location used to facilitate the localization of sensors
+and co-registration, anatomical landmarks are locations on a research subject such as the nasion
+(for a detailed definition see the coordinate system description in the BIDS specification).
+The `*_coordsystem.json` is REQUIRED if the optional `*_optodes.tsv` is specified. If a corresponding
+anatomical MRI is available, the locations of anatomical landmarks in that scan should also be stored
+in the `*_T1w.json` file which goes alongside the fNIRS data.
+
+Not all fNIRS systems provide 3D coordinate information or digitization capabilities.
+In this case, only x and y are specified and z is `"n/a"`.
+
+Fields relating to the fNIRS optode positions:
+
+{{ MACROS___make_sidecar_table(["nirs.CoordinateSystem", "nirs.CoordinateSystemDescriptionRec"]) }}
```

Comment:
@effigies @tsalo The enumerated types in this table are very long. Should this go back to the old way of doing things? Is there a change that should be made to the macro?

Reply:
I wouldn't be opposed to adding a new parameter like `n_values_to_combine` that would replace the enumerated values with a sentence like "For a list of valid values for this field, see the associated glossary entry." And the "associated glossary entry" text would link to the glossary entry. I can take a crack at that.

Reply:
@rwblair WDYT of #1260?

Reply:
It looked good to me. I looked at some of the collapse styles in mkdocs material, but for the simplicity and pdf rendering I think this is a good option.